python/lsst/pipe/base/connectionTypes.py

```diff
@@ -194,6 +194,12 @@ class BaseInput(DimensionedConnection):
     """
     deferLoad: bool = False
 
+    optional: ClassVar[bool] = False
+    """Quanta are not generated and never (fully) executed when there are no
+    datasets for a non-optional input connection, and most input connections
+    cannot be made optional.
+    """
+
```

Comment:
I want to have a consistent interface for `BaseInput`, where users can read `.optional` without checking whether they have an `Input` or `PrerequisiteInput`, because we mix those in so many contexts.  The alternative was
```
@property
def optional(self) -> bool:
    return False
```
in the base class and then overriding that to be backed by a `_optional` attribute in `PrerequisiteInput` only, and this seemed cleaner.